flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/TestingSchedulingTopology.java

```diff
@@ -140,6 +143,20 @@ private void addSchedulingResultPartitions(
         }
     }
 
+    private void addSchedulingResultPartitions(
+            final Iterable<ConsumedPartitionGroup> resultPartitionGroups,
+            final Function<IntermediateResultPartitionID, SchedulingResultPartition>
+                    resultPartitionById) {
+        for (ConsumedPartitionGroup consumedPartitionGroup : resultPartitionGroups) {
+            for (IntermediateResultPartitionID consumedPartitionId : consumedPartitionGroup) {
+                schedulingResultPartitions.put(
+                        consumedPartitionId,
+                        (TestingSchedulingResultPartition)
```

Comment:
by changing `Function<IntermediateResultPartitionID, SchedulingResultPartition>` to `Function<IntermediateResultPartitionID, TestingSchedulingResultPartition>` this force conversion can be avoided.

Reply:
We can remove this method since there is no need of it.